packages/monorepo-builder/src/VersionValidator.php

```diff
@@ -79,12 +84,14 @@ private function appendAppendingComposerJson(array $packageVersionsPerFile): arr
             return $packageVersionsPerFile;
         }
 
+        $isYamlConfig = file_exists(self::MONOREPO_BUILDER_YAML);
+
         foreach ($appendingComposerJson->getRequire() as $packageName => $packageVersion) {
-            $packageVersionsPerFile[$packageName][self::MONOREPO_BUILDER_YAML] = $packageVersion;
+            $packageVersionsPerFile[$packageName][$isYamlConfig ? self::MONOREPO_BUILDER_YAML : self::MONOREPO_BUILDER_PHP] = $packageVersion;
         }
 
         foreach ($appendingComposerJson->getRequireDev() as $packageName => $packageVersion) {
-            $packageVersionsPerFile[$packageName][self::MONOREPO_BUILDER_YAML] = $packageVersion;
+            $packageVersionsPerFile[$packageName][$isYamlConfig ? self::MONOREPO_BUILDER_YAML : self::MONOREPO_BUILDER_PHP] = $packageVersion;
```

Comment:
Does this needs to be resolved in every loop?
What about putting it right under `$isYamlConfig` line above?

Reply:
I did, but it got error cs need to pass constant to the last array key

Reply:
see phpstan error at https://github.com/symplify/symplify/runs/1166029627 

```bash

 ------------------------------------------------------------------------
  packages/monorepo-builder/src/VersionValidator.php:96
 ------------------------------------------------------------------------
  - '#Constant monorepoBuilderConfig not found#'
```

Reply:
Great, that's exactly what I meant :+1: 

That's because of typo here:

```diff
-$packageVersionsPerFile[$packageName][monorepoBuilderConfig]
+$packageVersionsPerFile[$packageName][$monorepoBuilderConfig]
```

Reply:
Thank you. Fixed now.